ambry-network/src/test/java/com.github.ambry.network/SSLSelectorTest.java

```diff
@@ -203,6 +202,27 @@ public void testEmptyRequest()
     assertEquals("", blockingRequest(connectionId, ""));
   }
 
+  @Test
+  public void testSSLConnect()
+      throws IOException {
+    String connectionId =
+        selector.connect(new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE, PortType.SSL);
+    while (!selector.connected().contains(connectionId)) {
+      selector.poll(10000L);
+    }
+    Assert.assertTrue("Channel should have been ready by now ", selector.isChannelReady(connectionId));
+  }
+
+  @Test
+  public void testCloseDuringSslHandshake()
```

Comment:
This is not necessarily testing close during ssl handshake. This is just close after issuing connect call - at the time of this close - connection could be in any of the three states - not established, established but not handshake complete, established and handshake complete. In all three cases, the connection should get added to the `disconnected` list as per the code. So you just have to rename this to `testCloseAfterConnectCall()` or something.